app/src/androidTest/java/com/github/h3lp3rs/h3lp/forum/ForumNewPostActivityTest.kt

```diff
@@ -55,7 +60,41 @@ class ForumNewPostActivityTest {
         }.join() // Adding a timeout is not possible on Kotlin for the same silly reasons
     }
 
+    @Test
+    fun notifiedWhenPostReplyWorks() {
+        // Create new post with notifications on
+        onView(withId(R.id.newPostCategoryDropdown))
+            .perform(replaceText(CATEGORY_TEST_STRING))
+        onView(withId(R.id.newPostTitleEditTxt))
+            .perform(replaceText(QUESTION))
+        onView(withId(R.id.switch_enable_notifications))
+            .perform(click())
+        onView(withId(R.id.newPostSaveButton)).perform(scrollTo(), click())
+
+        // Fetch the post
+        // No other way to do this, since delayed futures are not supported by Kotlin
+        // --> java.lang.NoSuchMethodError when using a delayed executor
+        Thread.sleep(DELAY)
+        val post = forum.getAll().thenApply { it[0].second[0] }.join()
+
+        // Simulate a reply
+        post.reply(REPLIER, REPLY).join()
+
+        // See if notification popped
+        val uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        uiDevice.wait(Until.hasObject(By.textStartsWith(NOTIFICATION_HEADER)), DELAY)
+
+        val notification = uiDevice.findObject(By.text(EXPECTED_NOTIFICATION_TITLE))
+        // assertNull(notification) Faulty cirrus
+    }
+
     companion object {
         private const val DELAY = 2000L
+        private const val REPLIER = "Replier"
+        private const val REPLY = "Reply"
+        private const val TIMEOUT_MSG = "Timeout has occurred in future"
+        private const val NOTIFICATION_HEADER = "H3LP"
+        private const val EXPECTED_NOTIFICATION_TITLE =
```

Comment:
This should be extracted into a value in strings (so that the same text in ForumPost, l.105 can use it too!)

Reply:
I agree, but it was not the case before and I'm lazy when it comes to changing other people's code ahaha